src/prerequisites/signals.py

```diff
@@ -87,3 +88,32 @@ def update_cache_triggered_by_prereq(sender, instance, *args, **kwargs):
         # Cover this instance in a post_delete signal receiver for Quest objects.
         if instance.parent_object:
             update_conditions_for_quest.apply_async(kwargs={'quest_id': instance.parent_object.id, 'start_from_user_id': 1}, queue='default')
+
+
+@receiver(post_save, sender=Prereq)
+def on_quest_badge_save_with_rank_prereq(sender, instance, *args, **kwargs):
+    """ This function checks if rank was added as a prerequisite and doesn't exist as an initial object of CytoScape map,
+    then creates a new map.
+    """
+    if instance.parent_content_type.model not in ['quest', 'badge']:
+        return
+
+    # check if preq 'slot' (prereq/or_prereq) has the conditions neede to create Cyto map
+    # create map if so
+    def check_conditions_and_create_map(prereq_invert, prereq_content_type, prereq_object_id):
+        if prereq_invert or \
+                (prereq_content_type is None or prereq_content_type.model != 'rank') or \
+                CytoScape.objects.filter(initial_content_type=prereq_content_type, initial_object_id=prereq_object_id).exists():
+            return
+
+        CytoScape.objects.create(
```

Comment:
Map is being create with `CytoScape` create. But ill use ` generate_map()` just in case
![image](https://github.com/bytedeck/bytedeck/assets/39788517/223d1c64-3cd4-474e-9d70-9fa9334e6d36)
![image](https://github.com/bytedeck/bytedeck/assets/39788517/98cb6e07-ee23-4ae5-8d2e-38d9c6341f8d)

Reply:
aj just realized theres supposed to be a quest there. fixed